src/main/java/io/xdb/core/persistence/PersistenceTableRowToUpsert.java

```diff
@@ -0,0 +1,84 @@
+package io.xdb.core.persistence;
+
+import io.xdb.gen.models.AttributeWriteConflictMode;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.Getter;
+
+@Getter
+public class PersistenceTableRowToUpsert {
+
+    private final String tableName;
+    String primaryKeyColumnName = null;
+    String primaryKeyColumnValue = null;
+    /**
+     * key: column name
+     * value: column value
+     */
+    private final Map<String, String> otherColumns = new HashMap<>();
+    private final AttributeWriteConflictMode writeConflictMode;
+
+    private PersistenceTableRowToUpsert(final String tableName, final AttributeWriteConflictMode writeConflictMode) {
+        this.tableName = tableName;
+        this.writeConflictMode = writeConflictMode;
+    }
+
+    /**
+     * Create a table row to upsert in persistence.
+     *
+     * @param tableName             the name of the table.
+     * @param writeConflictMode     the mode to choose when there is a conflict in upsert.
+     * @return  the table row to upsert.
+     */
+    public static PersistenceTableRowToUpsert create(
+        final String tableName,
+        final AttributeWriteConflictMode writeConflictMode
+    ) {
+        return new PersistenceTableRowToUpsert(tableName, writeConflictMode);
+    }
+
+    /**
+     * Create a table row to upsert in persistence with the primary key.
+     *
+     * @param tableName             the name of the table.
+     * @param columnName            the column name of the primary key.
+     * @param columnValue           the column value of the primary key.
```

Comment:
looks like you are missing the context here -- from user's stand point, the global attribute value is not just a plain string value. It can be integer, datetime, object, etc. Just like all different database column types that you can use.

The xdb-apis only accepted string simply because this is the way for xdb-server to understand and make up the query to write into database. 

That's why this one is needed in golang sdk: https://github.com/xdblab/xdb-golang-sdk/blob/main/xdb/db_converter.go